build-tools/src/main/java/org/elasticsearch/gradle/plugin/GeneratePluginPropertiesTask.java

```diff
@@ -0,0 +1,146 @@
+
+    private static final String PROPERTIES_FILENAME = "plugin-descriptor.properties";
+
+    @Inject
+    public GeneratePluginPropertiesTask(ProjectLayout projectLayout) {
+        setDescription("Generate " + PROPERTIES_FILENAME);
+        getOutputFile().convention(projectLayout.getBuildDirectory().file("generated-descriptor/" + PROPERTIES_FILENAME));
+    }
+
+    @Input
+    public abstract Property<String> getPluginName();
+
+    @Input
+    public abstract Property<String> getPluginDescription();
+
+    @Input
+    public abstract Property<String> getPluginVersion();
+
+    @Input
+    public abstract Property<String> getElasticsearchVersion();
+
+    @Input
+    public abstract Property<String> getJavaVersion();
+
+    @Input
+    public abstract Property<String> getClassname();
+
+    @Input
+    public abstract ListProperty<String> getExtendedPlugins();
+
+    @Input
+    public abstract Property<Boolean> getHasNativeController();
+
+    @Input
+    public abstract Property<Boolean> getRequiresKeystore();
+
+    @Input
+    public abstract Property<PluginType> getPluginType();
+
+    @Input
+    public abstract Property<String> getJavaOpts();
+
+    @Input
+    public abstract Property<Boolean> getIsLicensed();
+
+    @InputFiles
+    public abstract ConfigurableFileCollection getModuleInfoFile();
+
+    @OutputFile
+    public abstract RegularFileProperty getOutputFile();
+
+    @TaskAction
+    public void generatePropertiesFile() throws IOException {
+        Map<String, Object> props = new HashMap<>();
+        props.put("name", getPluginName().get());
+        props.put("description", getPluginDescription().get());
+        props.put("version", getPluginVersion().get());
+        props.put("elasticsearchVersion", getElasticsearchVersion().get());
+        props.put("javaVersion", getJavaVersion().get());
+        PluginType pluginType = getPluginType().get();
+        props.put("classname", pluginType.equals(PluginType.BOOTSTRAP) ? "" : getClassname().get());
+        props.put("extendedPlugins", String.join(",", getExtendedPlugins().get()));
+        props.put("hasNativeController", getHasNativeController().get());
+        props.put("requiresKeystore", getRequiresKeystore().get());
+        props.put("type", pluginType.toString());
+        props.put("javaOpts", getJavaOpts().get());
+        props.put("licensed", getIsLicensed().get());
+        props.put("modulename", findModuleName());
+
+        SimpleTemplateEngine engine = new SimpleTemplateEngine();
+        Path outputFile = getOutputFile().get().getAsFile().toPath();
+        Files.createDirectories(outputFile.getParent());
+        try (
+            var inputStream = GeneratePluginPropertiesTask.class.getResourceAsStream("/" + PROPERTIES_FILENAME);
+            var reader = new BufferedReader(new InputStreamReader(inputStream));
+            var writer = Files.newBufferedWriter(outputFile, StandardCharsets.UTF_8)
```

Comment:
We should be creating this writer in a try-with-resources. As it is we are never closing this writer.

Reply:
Same probably goes for the reader.

Reply:
These are in a try-with-resources already?

Reply:
I'm a moron that cannot differentiate a `(` from a `{`. Disregard.